### PR TITLE
feat(CSI): update frame and area objects

### PR DESCRIPTION
**Pull request by connorivy**
## Changes:

- set object guids for frame and area objects on receive
- collect all existing project guids before receive and compare incoming application Ids to existing guids before creating new element
- add updating functionality for frames and area objects
- tell converter which operation is happening to avoid always doing operations needed for sending
- break send operation into smaller methods in order be able to call from the receive operation if needed (for future diffing purposes)
- fix stream state loading and saving
- add support for etabs 20.3.0

## Screenshots:
ezgif-5-7cc50eb100
ezgif-2-11051240ef

## Validation of changes:
- manual receiving different commits with the same objects in different locations from Revit

## Checklist:

- [x] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [x] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [x] My commits are related to the pull request and do not amend unrelated code or documentation.
- [x] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

File: ConnectorCSI/ConnectorCSIShared/UI/ConnectorBindingsCSI.Send.cs [hunk 1]
```diff
@@ -148,12 +175,26 @@ public override async Task<string> SendStream(StreamState state, ProgressViewMod
       if (progress.Report.OperationErrorsCount != 0)
         return null;
 
+      if (branchName != null)
+      {
+        var branchesSplit = state.BranchName.Split('/');
+        branchesSplit[branchesSplit.Count() - 1] = branchName;
+        branchName = string.Join("", branchesSplit);
```
Comment (teocomi) [inline ConnectorCSI/ConnectorCSIShared/UI/ConnectorBindingsCSI.Send.cs:182]: why is this needed?

Comment (connorivy) [reply ConnectorCSI/ConnectorCSIShared/UI/ConnectorBindingsCSI.Send.cs:182]: that was sort of left over from the structural diffing that I did. I was sending to a "report" branch and I wanted it to have the same nesting as the branch that I was receiving from. I figure we may use that functionality eventually so I would just leave it. Let me know and I can get rid of that.

**Review by teocomi — APPROVED** [review]
